nvtabular/ops/data_schema.py

```diff
@@ -0,0 +1,173 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+import os
+
+import dask.dataframe as dd
+from nvtx import annotate
+
+from nvtabular.dispatch import DataFrameType
+
+from ..column import Columns
+from .base import Operator
+from .stat_operator import StatOperator
+
+LOG = logging.getLogger("nvtabular")
+
+
+class DataSchema(StatOperator):
+    SCHEMA_FILE_NAME = "schema.pb"
+
+    def __init__(self, tags_by_column, output_path=None):
+        super().__init__()
+        self.schema_path = os.path.join(output_path, self.SCHEMA_FILE_NAME) if output_path else None
+        self.col_names = []
+        self.col_types = []
+        self.col_dtypes = []
+        self.tags_by_column = tags_by_column
+        self.schema = None
+
+    @classmethod
+    def calculate_on_dataset(cls, dataset, tags_by_column, output_path=None, client=None):
+        from nvtabular.column_group import ColumnGroup
+        from nvtabular.workflow import Workflow
+
+        stats = cls(tags_by_column, output_path=output_path)
+
+        col_group = ColumnGroup([])
+
+        for column in dataset.columns:
+            col_group += ColumnGroup(column, tags=tags_by_column.get(column))
+
+        workflow = Workflow(col_group >> stats, client=client)
+        workflow.fit(dataset)
+
+        return stats.schema
+
+    @classmethod
+    def load(cls, directory):
+        from tensorflow_metadata.proto.v0 import schema_pb2
+
+        schema_file = os.path.join(directory, cls.SCHEMA_FILE_NAME)
+
+        LOG.info("Loading schema from %s", schema_file)
+
+        if not os.path.exists(schema_file):
+            return None
+
+        schema = schema_pb2.Schema()
+        with open(schema_file, "rb") as f:
+            schema.ParseFromString(f.read())
+
+        return schema
+
+    def transform(self, columns: Columns, df: DataFrameType) -> DataFrameType:
+        return df
```

Comment:
This method seems extraneous unless maybe the super-class requires it to implemented?

Reply:
Good catch!